src/PortingAssistant.Client.Common/Utils/FileSystemAccess.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Security.Principal;
+using System.Security.AccessControl;
+
+namespace PortingAssistant.Client.Common.Utils
+{
+    public static class FileSystemAccess
+    {
+        /// <summary>
+        /// Checks directory and all content for write access
+        /// </summary>
+        /// <param name="path">Directory path</param>
+        /// <returns>List of items without write access</returns>
+        public static List<string> CheckWriteAccessForDirectory(string path)
+        {
+            var result = new List<string>();
+            foreach (string file in Directory.GetFiles(path))
+            {
+                if (!HaveFileWriteAccess(file))
+                {
+                    result.Add(file);
+                }
+            }
+            foreach (string subDirectory in Directory.GetDirectories(path))
+            {
+                if (!HaveDirectoryWriteAccess(subDirectory))
+                {
+                    result.Add(subDirectory);
+                }
+                result.AddRange(CheckWriteAccessForDirectory(subDirectory));
+            }
+            return result;
+        }
+
+        private static bool HaveFileWriteAccess(string path)
+        {
+            var fileInfo = new FileInfo(path);
+            FileSecurity accessControl = fileInfo.GetAccessControl();
+            AuthorizationRuleCollection rules = accessControl.GetAccessRules(true, true, typeof(SecurityIdentifier));
+            return CheckAuthorizationRules(rules);
+        }
+
+        private static bool HaveDirectoryWriteAccess(string path)
+        {
+            var directoryInfo = new DirectoryInfo(path);
+            DirectorySecurity accessControl = directoryInfo.GetAccessControl();
+            AuthorizationRuleCollection rules = accessControl.GetAccessRules(true, true, typeof(SecurityIdentifier));
+            return CheckAuthorizationRules(rules);
+        }
```

Comment:
Did you try these in a scenario where the user has write access to the machine but the folder is restricted (For example, a user who has access to their local user folder but is trying to write to another drive)

Reply:
I didn't try that particular scenario. I did test limiting access to a specific folder within a project. Given that we are now  trying to access each file/directory, even if we are trying to write to another drive assuming the file system paths are valid it should be fine I think? I will try and find a way to test this though.

That does raises another question of should I be concerned with paths that are not valid?